hyperledger/besu
# Fix tracing with reference tests

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>




## PR description

This PR fixes several bugs discovered thanks to the reference tests and goerli (300,000 blocks checked)

Fixes traces in the following cases : 
 -  Contract creation colission
 - Stack underflow during contract creation
 - refund_CallToSuicideTwice 
 - Missing op in VmTrace 
 - Missing reverted error message in the first frame (If there is no other CALL)
 - GasRefund missing on some transaction. This resulted in an invalid gasUsed

This PR also allows you to trace only what you need (no need to capture memory or storage during tracing). The capture will depend on the options chosen by the user (vmTrace, trace, etc.) And this can improve performance

## Tested

- Check with 300,000 blocks on goerli
- Check with tests already present on besu
- Check with blocks sent by Alethio
